### PR TITLE
Set AllowDirect for upgrades from prior to 2.9.x

### LINKED ISSUE #1208: Error with CreateKeyValue when upgrading from 2.8.x to 2.9.x
## Defect

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [x] Included nats.go version
 - [x] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats.go` and the `nats-server` if one was involved:
nats.go: v1.23.0
nats-server: v2.9.14

#### OS/Container environment:
Ubuntu

#### Steps or code to reproduce the issue:
Create a KV bucket while running nats-server 2.8.4. e.g
```
kv, err := mgr.CreateKeyValue(&nats.KeyValueConfig{
		Bucket:      "test-bucket",
		Description: "bucket to demo defectt",
	})
```

Upgrade server to 2.9.x

Run the same code again

#### Expected result:
`CreateKeyValue`  returns the existing bucket

#### Actual result:
Error returned: ` nats: stream name already in use`

### PR DESCRIPTION
**Pull request by dbason**
Fixes #1208

### CODE REVIEW

**Comment by dbason** [comment]
I reviewed the test failure but can't see a reason for it.  Is it possible this is a flake?

File: kv.go [hunk 1]
```diff
@@ -446,11 +446,15 @@ func (js *js) CreateKeyValue(cfg *KeyValueConfig) (KeyValue, error) {
...
 		// and the only difference is the discard policy, then update
 		// the stream.
+		// The same logic applies for clusters created pre 2.9.x and
+		// the AllowDirect setting
 		if err == ErrStreamNameAlreadyInUse {
 			if si, _ = js.StreamInfo(scfg.Name); si != nil {
 				// To compare, make the server's stream info discard
 				// policy same than ours.
 				si.Config.Discard = scfg.Discard
+				// Allso need to set allow direct for v2.9.x+
```
Comment (piotrpio) [inline kv.go:456]:
```suggestion
				// Also need to set allow direct for v2.9.x+
```

File: kv.go [hunk 2]
```diff
@@ -446,11 +446,15 @@ func (js *js) CreateKeyValue(cfg *KeyValueConfig) (KeyValue, error) {
 		// and we are now moving to a v2.7.2+. If that is the case
 		// and the only difference is the discard policy, then update
 		// the stream.
+		// The same logic applies for clusters created pre 2.9.x and
+		// the AllowDirect setting
```
Comment (piotrpio) [inline kv.go:450]:
```suggestion
		// The same logic applies for KVs created pre 2.9.x and
		// the AllowDirect setting.
```

**Review by piotrpio — COMMENTED** [review]
Thanks for the PR! It looks good, I just have some minor suggestions.

The test seems to be flaky indeed, I'll re-run the job to be sure.

**Review by piotrpio — APPROVED** [review]
LGTM!

**Pull request merged**